### ISSUE TITLE
AppImage doesn't quite run in Arch Linux

### ISSUE THREAD

**Issue by x80486** [Cloaklet/Cloak#1]
I was searching for a `gocryptfs`'s UI and I found this one. I decided to give it a try, but looks like the AppImage doesn't quite run in Arch Linux:

```shell
Oct 25 23:36:04 uplink systemd[541]: tmp-.mount_Cloak\x2dsZdf7s.mount: Succeeded.
Oct 25 23:36:04 uplink systemd[1]: tmp-.mount_Cloak\x2dsZdf7s.mount: Succeeded.
Oct 25 23:36:18 uplink gnome-shell[1080]: libinput error: event20 - Logitech M525: client bug: event processing lagging behind by 23ms, your system is too slow
Oct 25 23:36:18 uplink com.bitwarden.desktop.desktop[2648]: Xlib:  extension "MIT-SCREEN-SAVER" missing on display ":99.0".
Oct 25 23:36:38 uplink systemd[541]: tmp-.mount_Cloak\x2dek6Zxv.mount: Succeeded.
Oct 25 23:36:38 uplink systemd[1]: tmp-.mount_Cloak\x2dek6Zxv.mount: Succeeded.
Oct 25 23:36:40 uplink systemd[541]: app-gnome-gnome\x2dsystem\x2dmonitor-845434.scope: Succeeded.
```

As you can see, I'm able to start it, but that's about it. I don't see anything (UI) running.

> I have `gocryptfs` installed as a package on its own.

**Comment by JokerQyou** [comment]
Hmmm, I have no access to an Arch Linux machine, but Cloak does run on Linux Mint (which is a variant of Ubuntu). The app starts as a tray icon (or menu icon) with a menu to open the UI in your default browser, see the screenshot below.

<img width="86" alt="Tray icon" src="https://user-images.githubusercontent.com/1465267/97403761-f933a200-192f-11eb-8a9d-676dbb8a5e47.png">

If the process does not crash and you are not seeing any tray icon, maybe there's just something wrong with the icon. Please try manually navigating to `http://127.0.0.1:9763/` in your browser. The UI is shown below.

![UI](https://user-images.githubusercontent.com/1465267/97404032-66dfce00-1930-11eb-968f-7c9917b13c66.png)

**Comment by JokerQyou** [comment]
Also you can find Cloak's data in `~/.cloaklet.cloak` on Linux. Its log is stored in `~/.cloaklet.cloak/logs`.

**Comment by x80486** [comment]
OK, I didn't know any of that :sweat_smile:

It's working fine: `0.7.0 (240f7ac)`. The only problem right now is that the mount point is not found in `Files` (under `GNOME`). So I have to look in the logs where the mount-point is and use it. What I've see with some FUSE-related applications (and the same `gocryptfs`) is that they reveal the mounted directory in the operating system's file browser itself, so it's probably doable here also.

I know Linux have always something new and there are like three million distros, but supporting the main DEs (`GNOME` and `KED`) out-of-the-box first should be a good route.

Thanks for the work on `Cloak`! Feel free to close this one :sunglasses:

**Comment by JokerQyou** [comment]
I'm glad you figured it out! On macOS I did implement the "reveal in OS file manager" logic, but with Linux I have no idea which file manager to support. Personally I'm on Linux Mint with XFCE, and as far as I understand, each desktop environment ships its own file manager. I'll look into this in a few days.

**Comment by JokerQyou** [comment]
> ... The only problem right now is that the mount point is not found in `Files` (under `GNOME`). So I have to look in the logs where the mount-point is and use it. What I've see with some FUSE-related applications (and the same `gocryptfs`) is that they reveal the mounted directory in the operating system's file browser itself, so it's probably doable here also.

I've checked the code and ran the latest released version (0.7.0), the "reveal in OS file manager" feature works (on my machine). It uses `xdg-open` to do this. Could you kindly try clicking "Reveal Drive" button on the right panel after successfully unlocking a vault? Does it do anything? If not, could you also check `which xdg-open` in your terminal?

**Comment by x80486** [comment]
I don't have `xdg-open`, but indeed, as soon as it's installed, it works fine! :partying_face:
